src/main/java/com/dell/pravega/perf/PerfStats.java

```diff
@@ -0,0 +1,347 @@
+            int index = 0;
+
+            for (int i = 0; i < percentiles.length; i++) {
+                percentileIds[i] = (int) (count * percentiles[i]);
+            }
+
+            for (LatencyRange lr : latencyRanges) {
+                while ((index < percentileIds.length) &&
+                        (lr.start <= percentileIds[index]) && (percentileIds[index] <= lr.end)) {
+                    values[index++] = lr.latency;
+                }
+            }
+            return values;
+        }
+
+        public void record(int bytes, int latency) {
+            totalBytes += bytes;
+            latencies[latency]++;
+        }
+
+        public void record(long start, int bytes, int latency) {
+            this.record(bytes, latency);
+        }
+
+        public void printTotal(long endTime) {
+            countLatencies();
+            final double elapsed = (endTime - start) / 1000.0;
+            final double recsPerSec = count / elapsed;
+            final double mbPerSec = (this.totalBytes / (1024.0 * 1024.0)) / elapsed;
+            int[] percs = getPercentiles();
+
+            System.out.printf(
+                    "%d records %s, %.3f records/sec, %d bytes record size, %.2f MB/sec, %.1f ms avg latency, %.1f ms max latency" +
+                            ",%d ms 50th, %d ms 75th, %d ms 95th, %d ms 99th, %d ms 99.9th\n",
+                    count, action, recsPerSec, messageSize, mbPerSec, totalLatency / ((double) count), (double) maxLatency,
+                    percs[0], percs[1], percs[2], percs[3], percs[4]);
+        }
+    }
+
+    private class CSVLatencyWriter extends LatencyWriter {
+        final private String csvFile;
+        final private CSVPrinter csvPrinter;
+
+        public CSVLatencyWriter(String action, int messageSize, long start, String csvFile) throws IOException {
+            super(action, messageSize, start);
+            this.csvFile = csvFile;
+            csvPrinter = new CSVPrinter(Files.newBufferedWriter(Paths.get(csvFile)), CSVFormat.DEFAULT
+                    .withHeader("Start Time (Milliseconds)", "event size (bytes)", action + "Latency (Milliseconds)"));
+        }
+
+        private void readCSV() {
+            try {
+                CSVParser csvParser = new CSVParser(Files.newBufferedReader(Paths.get(csvFile)), CSVFormat.DEFAULT
+                        .withFirstRecordAsHeader().withIgnoreHeaderCase().withTrim());
+
+                for (CSVRecord csvEntry : csvParser) {
+                    record(Integer.parseInt(csvEntry.get(1)), Integer.parseInt(csvEntry.get(2)));
+                }
+                csvParser.close();
+            } catch (IOException ex) {
+                ex.printStackTrace();
+            }
+        }
+
+        @Override
+        public void record(long start, int bytes, int latency) {
+            try {
+                csvPrinter.printRecord(start, bytes, latency);
+            } catch (IOException ex) {
+                ex.printStackTrace();
```

Comment:
This isn't a good way to deal with errors. Perhaps a logger would be best. Alternatively if you think this shouldn't occur, you could convert it to a run-time exception an re-throw.

Reply:
yes , its changed now